dash/development/base_component.py

```diff
@@ -392,6 +402,14 @@ def __repr__(self):
             props_string = repr(getattr(self, "children", None))
         return f"{self._type}({props_string})"
 
+    @classmethod
+    def _get_base_nodes(cls):
+        if not cls._base_nodes:
```

Comment:
if a component has lots of props but none of them accept components, will we run this iteration every time one of them is instantiated? If so, can we avoid that? Moving `+ ["children"]` in here would be one way perhaps?

Reply:
oh sorry, it's only iterating over `_children_props` - so the problem only comes if there are lots of nested props and no base props. Still would be best to avoid.

Reply:
Let's move that to the generator.

Reply:
Added a legacy test from the old standard component output.